bindings/distrdf/python/DistRDF/HeadNode.py

```diff
@@ -636,23 +636,40 @@ def __init__(self, backend: BaseBackend, npartitions: Optional[int], localdf: RO
         super().__init__(backend, npartitions, localdf)
 
         # Information about friend trees, if they are present.
+        self.sampleMap: Dict[str, Ranges.SerializableRSample] = None 
+        
         self.friendinfo: Optional[ROOT.Internal.TreeUtils.RFriendInfo] = None
-
+                
+            
         # Retrieve the RDatasetSpec that will be processed
         if isinstance(args[0], ROOT.RDF.Experimental.RDatasetSpec):
             # RDataFrame(rdatasetspec)
             fi = args[0].GetFriendInfo()
             self.friendinfo = fi if not fi.fFriendNames.empty() else None
+
         else:
             raise RuntimeError(
                 f"First argument {args[0]} of type {type(args[0])} is not supported "
                 "in RDatasetSpecHeaNode")
-
+                
+        if (args[0].GetEntryRangeBegin() != 0):
+            warnings.warn("Setting global range is not supported in distributed parsing of RDatasetSpec, the set range will be ignored and all events will be processed.")
+            
         # subtreenames: names of all subtrees in the chain or full path to the tree in the file it belongs to
-        self.subtreenames = [str(treename)
-                             for treename in args[0].GetTreeNames()]
-        self.inputfiles = [str(filename)
-                           for filename in args[0].GetFileNameGlobs()]
+        self.subtreenames = [str(treename) for treename in args[0].GetTreeNames()]
+        self.inputfiles = [str(filename) for filename in args[0].GetFileNameGlobs()]
+        
+        self.samples = [ROOT.RDF.Experimental.RSample(sample) for sample in ROOT.Internal.RDF.MoveOutSamples(args[0])]
+        
+        # A map to map file id (filename/tree) to SerializableRSamples
+        self.sampleMap = {}
+        
+        for sample in self.samples:
+            trees = sample.GetTreeNames()
+            files = sample.GetFileNameGlobs()
+            for i in range (len(files)):
+                sampleId = files[i] + "/" + trees[i]
+                self.sampleMap.update({sampleId : Ranges.SerializableRSample(sample)})
```

Comment:
Since we're anyway just storing `self.sampleMap`, shouldn't we remove it `self.samples` and just use it as a local variable? In that case, it wouldn't even be necessary to convert it to a list, just the vector returned from `MoveOutSamples` should be good.